Task3/Task3_Atanas_Kisyov/src/python/car_wash/db_connection/connection_pools.py

```diff
@@ -0,0 +1,101 @@
+"""Read-only and read/write database connection pools definitions"""
+from collections import deque
+import logging
+import time
+
+import psycopg2
+
+from python.car_wash.db_connection.pool_base_class import PoolBaseClass
+
+
+class ReadOnlyPool(PoolBaseClass):
+
+    """Read-only connection pool"""
+
+    def __init__(self, hostname, database_name, pool_size: int) -> None:
+        """Initialize read-only pool"""
+
+        self.hostname = hostname
+        self.database_name = database_name
+        self._pool = deque()
+        self.pool_size = pool_size
+        self._open_connection = None
+    
+    def __enter__(self) -> psycopg2.connect:
```

Comment:
Please replace __enter__ and __exit__ into my last example in chat

Reply:
Done.